api-client/src/utils/geoJson.ts

```diff
@@ -8,17 +8,26 @@ export const embedInCollection = (features: PoiFeature[]): PoiFeatureCollection
   features,
 })
 
-export const prepareFeatureLocation = (poi: PoiModel, userLocation: LocationType | null): PoiFeature | null => {
+export const prepareFeatureLocation = (poi: PoiModel, userLocation: LocationType | null): PoiFeature => {
   const { featureLocation } = poi
-  if (userLocation && featureLocation) {
-    const distanceValue = distance(userLocation, featureLocation.geometry.coordinates).toFixed(1)
+  if (userLocation) {
+    const distanceValue = distance(userLocation, featureLocation.geometry.coordinates).toFixed(0)
     return { ...featureLocation, properties: { ...featureLocation.properties, distance: distanceValue } }
   }
   return featureLocation
 }
 
-export const prepareFeatureLocations = (pois: Array<PoiModel>, userLocation: LocationType | null): PoiFeature[] =>
-  pois
-    .map(poi => prepareFeatureLocation(poi, userLocation))
-    .filter((feature): feature is PoiFeature => !!feature)
-    .sort((poi1: PoiFeature, poi2: PoiFeature) => poi1.properties.title.localeCompare(poi2.properties.title))
+export const prepareFeatureLocations = (pois: Array<PoiModel>, userLocation: LocationType | null): PoiFeature[] => {
+  const poiFeatures = pois.map(poi => prepareFeatureLocation(poi, userLocation))
+
+  if (userLocation) {
+    // sort by distance to current location
+    return poiFeatures.sort(
+      (poi1: PoiFeature, poi2: PoiFeature) =>
+        parseFloat(poi1.properties.distance ?? '0') - parseFloat(poi2.properties.distance ?? '0')
+    )
+  }
+  return poiFeatures.sort((poi1: PoiFeature, poi2: PoiFeature) =>
+    poi1.properties.title.localeCompare(poi2.properties.title)
+  )
```

Comment:
No need for the type annotations:
```suggestion
   if (userLocation) {
    // sort by distance to current location
    return poiFeatures.sort(
      (poi1, poi2) => parseFloat(poi1.properties.distance ?? '0') - parseFloat(poi2.properties.distance ?? '0')
    )
  }
  return poiFeatures.sort((poi1, poi2) => poi1.properties.title.localeCompare(poi2.properties.title))
```